src/envoy/http/service_control/handler_utils.cc

```diff
@@ -177,25 +177,35 @@ void fillLatency(const Envoy::StreamInfo::StreamInfo& stream_info,
   if (stream_info.requestComplete()) {
     latency.request_time_ms =
         convertNsToMs(stream_info.requestComplete().value());
-    filter_stats.filter_.request_time_.recordValue(latency.request_time_ms);
   }
 
   auto start = stream_info.firstUpstreamTxByteSent();
```

Comment:
The logic here is messy, I'll try to explain:
- If we know how long the backend took from request/response: overhead = total - backend
- Otherwise, we calculate overhead (less accurately) and calculate: backend = total - overhead